lua/nvim-treesitter/parsers.lua

```diff
@@ -1,17 +1,255 @@
+    url = "https://github.com/tree-sitter/tree-sitter-html",
+    files = { "src/parser.c", "src/scanner.cc" },
+  }
+}
+
+list.julia = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-julia",
+    files = { "src/parser.c", "src/scanner.c" },
+  }
+}
+
+list.json = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-json",
+    files = { "src/parser.c" },
+  }
+}
+
+list.css = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-css",
+    files = { "src/parser.c", "src/scanner.c" },
+  }
+}
+
+list.ocaml = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-ocaml",
+    files = { "src/parser.c", "src/scanner.cc" },
+  }
+}
+
+list.swift = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-swift",
+    files = { "src/parser.c" },
+  }
+}
+
+list.c_sharp = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-c-sharp",
+    files = { "src/parser.c", "src/scanner.c" },
+  },
+  filetype = 'cs'
+}
+
+list.typescript = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-typescript",
+    files = { "src/parser.c", "src/scanner.c" },
+    location = "tree-sitter-typescript/typescript"
+  }
+}
+
+list.tsx = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-typescript",
+    files = { "src/parser.c", "src/scanner.c" },
+    location = "tree-sitter-tsx/tsx"
+  },
+  filetype = 'typescriptreact'
+}
+
+list.scala = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-scala",
+    files = { "src/parser.c", "src/scanner.c" },
+  }
+}
+
+list.haskell = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-haskell",
+    files = { "src/parser.c", "src/scanner.cc" },
+  }
+}
+
+list.markdown = {
+  install_info = {
+    url = "https://github.com/ikatyang/tree-sitter-markdown",
+    files = { "src/parser.c", "src/scanner.cc" },
+  }
+}
+
+list.toml = {
+  install_info = {
+    url = "https://github.com/ikatyang/tree-sitter-toml",
+    files = { "src/parser.c", "src/scanner.c" },
+  }
+}
+
+list.vue = {
+  install_info = {
+    url = "https://github.com/ikatyang/tree-sitter-vue",
+    files = { "src/parser.c", "src/scanner.cc" },
+  }
+}
+
+list.elm = {
+  install_info = {
+    url = "https://github.com//razzeee/tree-sitter-elm",
+    files = { "src/parser.c", "src/scanner.cc" },
+  }
+}
+
+list.yaml = {
+  install_info = {
+    url = "https://github.com/ikatyang/tree-sitter-yaml",
+    files = { "src/parser.c", "src/scanner.cc" },
+  }
+}
+
+list.nix = {
+  install_info = {
+    url = "https://github.com/cstrahan/tree-sitter-nix",
+    files = { "src/parser.c", "src/scanner.cc" },
+  }
+}
+
+list.regex = {
+    install_info = {
+        url = "https://github.com/tree-sitter/tree-sitter-regex",
+        files = { "src/parser.c" }
+    }
+}
+
+local M = {
+  list = list
+}
+
+local ft_to_parsername = {}
```

Comment:
This dictionary should be made somehow accessible for configuration for users. In case, they want to use a parser from a mainstream language for niche-dialect.

Reply:
Maybe the users could setup their own parsers ? they don't need to access this particular variable if they setup new parsers directly in the `list`

Reply:
well this should not be available actually, because you cannot setup new parsers just editing this dict. I'm going to add some code to make this possible though.